alternator/server.cc

```diff
@@ -153,19 +153,35 @@ void server::set_routes(routes& r) {
     r.add(operation_type::POST, url("/"), handler);
 }
 
-future<> server::init(net::inet_address addr, uint16_t port) {
+future<> server::init(net::inet_address addr, std::optional<uint16_t> port, std::optional<uint16_t> tls_port, std::optional<tls::credentials_builder> creds) {
```

Comment:
Instead of two `std::optional<uint16_t>` port parameters, why not have just one port parameter and another parameter with the kind of connection to open?

Reply:
I didn't assume that these are exclusive